pycheribuild/projects/syzkaller.py

```diff
@@ -62,95 +68,161 @@ def setup_config_options(cls, **kwargs):
+        with self.set_env(PATH=self.get_path(), SOURCEDIR=self.cheribsd_dir):
             self.run_make("extract", parallel=False, cwd=self.gosrc)
             self.run_make("generate", parallel=False, cwd=self.gosrc)
 
+    def get_install_files(self, dir):
+        file_paths = []
+        for path in dir.iterdir():
+            if path.is_dir():
+                sub_files = self.get_install_files(path)
+                file_paths.extend(sub_files)
+            else:
+                file_paths.append(path)
+        return file_paths
+
     def install(self, **kwargs):
         # XXX-AM: should have a propert install dir configuration
-        native_build = self.source_dir / "bin"
-        mips64_build = native_build / "freebsd_mips64"
-        syz_remote_install = self.syzkaller_install_path() / "freebsd_mips64"
+        build_output_path = self.source_dir / "bin"
+        sdk_install_path = self.syzkaller_install_path()
 
-        self.makedirs(syz_remote_install)
+        self.makedirs(sdk_install_path)
 
-        self.install_file(native_build / "syz-manager", self.syzkaller_binary(), mode=0o755)
+        self.install_file(build_output_path / "syz-manager", self.syzkaller_binary(), mode=0o755)
 
         if not self.config.pretend:
-            # mips64_build does not exist if we preted, so skip
-            for fname in os.listdir(str(mips64_build)):
-                fpath = mips64_build / fname
+            # build does not exist if we preted, so skip
+            for fpath in self.get_install_files(build_output_path):
                 if os.path.isfile(fpath):
-                    self.install_file(fpath, syz_remote_install / fname, mode=0o755)
+                    self.install_file(fpath, sdk_install_path / fpath.relative_to(build_output_path),
+                                      mode=0o755)
 
     def clean(self) -> ThreadJoiner:
         self.run_cmd(["chmod", "-R", "u+w", self.build_dir])
-        self.make_args.set_env(
-            HOSTARCH="amd64",
-            TARGETARCH=self.crosscompile_target.cpu_architecture.value,
-            TARGETOS="freebsd",
-            GOROOT=self.goroot.expanduser(),
-            GOPATH=self.gopath.expanduser(),
-            CC=self.CC, CXX=self.CXX,
-            PATH=self._new_path)
 
         self.run_make("clean", parallel=False, cwd=self.gosrc)
         joiner = super().clean()
         return joiner
 
 
-class RunSyzkaller(SimpleProject):
-    target = "run-syzkaller"
+class BuildMorelloSyzkaller(BuildSyzkaller):
+    target = "morello-syzkaller"
+    github_base_url = "https://github.com/CTSRD-CHERI/"
+    repository = GitRepository(github_base_url + "cheri-syzkaller.git", force_branch=True,
+                               default_branch="morello-syzkaller")
+
+    def get_sdk_dir(self):
+        return self.config.morello_sdk_dir
+
+    def get_sdk_bindir(self):
+        return self.config.morello_sdk_bindir
+
+
+class VMSetting(Enum):
```

Comment:
TargetSytemSettings or something like that? Not all of these are virtual machines.

Reply:
I used vm for consistency. In syzkaller, vm is used to refer to them, and in the config file you specify the system settings under "vm". But I agree that TargetSystemSettings is more accurate, should I change it?